src/stacks/nestedBillingAlert.ts

```diff
@@ -0,0 +1,126 @@
+import * as cdk from '@aws-cdk/core';
+import * as cfn from '@aws-cdk/aws-cloudformation';
+import * as cw from '@aws-cdk/aws-cloudwatch';
+import * as b from '@aws-cdk/aws-budgets';
+
+import * as config from '../utils/config';
+
+import { NestedSNSStack } from './nestedSns';
+
+import { getTreatMissingData, getComparisonOperator } from '../utils/alarm';
+import { getDuration } from '../utils/metric';
+
+// Generate nested stack for billing alerts
+export class NestedBillingAlertStack extends cfn.NestedStack {
+  protected readonly snsStack: NestedSNSStack;
+
+  constructor(scope: cdk.Construct, id: string, snsStack: NestedSNSStack, props?: cfn.NestedStackProps) {
+    super(scope, id, props);
+
+    this.snsStack = snsStack;
+    const billingAlert = config.configGetBillingAlert() || null;
+
+    if (billingAlert && config.configIsEnabled(billingAlert)) {
+      Object.keys(billingAlert.alarm || {}).forEach(topic => {
+        const conf = billingAlert.alarm?.[topic];
+        if (conf && conf.enabled !== false) {
+          this.createAnomalyDetectionAlarm(id, billingAlert, topic);
+          this.createBudgetAlarm(id, conf, topic);
+        }
+      });
+    }
+  }
+
+  private createBudgetAlarm(id: string, alarmOpts: config.BillingAlertAlarmOptions, topic: string): void {
+    const budgetName = `${id}-${topic}-budget`;
+    const topicArn = cdk.Token.asString(this.snsStack.getTopicArn(topic));
+
+    const budgetParams: b.CfnBudgetProps = {
+      budget: {
+        budgetName: budgetName,
+        budgetType: 'COST',
+        budgetLimit: {
+          amount: alarmOpts.budgetLimit,
+          unit: 'USD',
```

Comment:
any reason why dollar is used as a unit?

Reply:
Yes, if trying something else I got the following error:
`Unable to create/update budget - EUR is not in the supported unit set: [USD]`
So I assuming only USD is supporter currently (I couldn't find any information in the docs).

Reply:
:thonkwtf: